Snowflake.Data/Core/FileTransfer/StorageClient/SFSnowflakeAzureClient.cs

```diff
@@ -330,13 +335,14 @@ public async Task DownloadFileAsync(SFFileMetadata fileMetadata, string fullDstP
             fileMetadata.resultStatus = ResultStatus.DOWNLOADED.ToString();
         }
 
-        private SFFileMetadata HandleFileHeaderErr(RequestFailedException ex, SFFileMetadata fileMetadata)
+        private SFFileMetadata HandleFileHeaderErr(Exception ex, SFFileMetadata fileMetadata)
```

Comment:
why the type got changed here? and...

Reply:
I think it's because the other client was using `Exception` and I copied the function parameters here without checking. I'll revert this change since, as you mentioned below, it's being casted to `RequestFailedException` anyway